react/react1/week3/todo-app3/src/components/ResultData.js

```diff
@@ -0,0 +1,39 @@
+import React, { useContext, useEffect } from 'react';
+import { UserContext } from './Context';
+
+function ResultData() {
+  const { list, setList, input, error, setError, isLoading, setIsLoading } =
+    useContext(UserContext);
+
+  const fetchData = async (result) => {
+    setIsLoading(true);
+    const res = await fetch(
+      `https://api.github.com/search/users?q=${result}`
+    ).then((res) => res.json());
+
+    if (res.message) {
+      setList([]);
+      setIsLoading(false);
+      return setError(true);
+    }
+    setList(res.items);
+    setError(false);
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    fetchData(input);
+  }, [input]);
+
+  return (
+    <div>
+      {list.map((item) => (
+        <p key={item.id}>{item.login}</p>
+      ))}
+      {isLoading && <p>Loading...</p>}
+      {error && <p>Sorry no item found</p>}
```

Comment:
The task was to show the cause of the error.
message "Sorry, no user found" should be show, when we have an empty array